### DISCUSSION TITLE
failing async test ends in timeout instead of go to catch clause

### DISCUSSION THREAD

**Discussion by sombriks** [avajs/ava#3311] [Q&A] [answered]
I have some async tests and everything goes ok except when i get some assertion failing. 

This is very similar to the issue described [here](https://github.com/avajs/ava/issues/2494#issuecomment-1134480337).

For example, the bellow snippet dies in timeout because the content-type is different to the expected:

```javascript
import test from 'ava'
import request from 'supertest'

import { app, database } from './main.js'

// other tests

test('should get htmx browser script', async t => {
  try {
    const result = await request(app.callback()).get('/htmx.js')
    t.regex(result.header["content-type"], /javascript/)
    t.is(200, result.status)
  } catch (e) {
    t.fail(e.message)
  }
})
```

But first i got the timeout and only then i get the message regarding the failing regex match:

![image](https://github.com/avajs/ava/assets/556695/54e22156-7563-4681-8344-7cbd6a047873)

Any guidance on how to avoid timeout is welcome. The sample code, ava version and everything else is in [this repo](https://github.com/sombriks/sample-htmx-koa).

**Comment by novemberborn** [comment, marked as answer]
The database isn't destroyed when the test fails. This code doesn't run after failures:

```js
test.after(async t => {
  await database.destroy()
})
```

Use the `.always` modifier instead:

```js
test.after.always(async t => {
  await database.destroy()
})
```

**Reply by sombriks** [reply]
Thanks a lot, it did solved the timeout issue, i can even get rid of the try/catch
